File: data/compilers_comparation/src/examples/first_example/src/func30.c
#include "first_example.h" 
array_t* func30(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      array_t* array16;
      if (pCounter > 0) {
         array16 = vars->data[--pCounter];
         array16->refC++;
         DEBUG_COPY(array16->id);
      } else {
         array16 = (array_t*)malloc(sizeof(array_t));
         array16->size = 84;
         array16->refC = 1;
         array16->id = 16;
         array16->data = (unsigned int*)malloc(array16->size*sizeof(unsigned int));
         memset(array16->data, 0, array16->size*sizeof(unsigned int));
         DEBUG_NEW(array16->id);
      }
      unsigned int loop8 = 0;
      unsigned int loopLimit8 = (rand()%loopsFactor)/4 + 1;
      for(; loop8 < loopLimit8; loop8++) {
         for (int i = 0; i < array16->size; i++) {
            array16->data[i]--;
         }
         for (int i = 0; i < array16->size; i++) {
            if (array16->data[i] == 5) { 
               return array16;
            }
         }
      }
      array_t_param params0;
      params0.size = 1;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      params0.data[0] = array16;
      array_t* array17 = func42(&params0, rng(), loopsFactor);
      DEBUG_RETURN(array17->id);
      free(params0.data);
      array17->refC--;
      if(array17->refC == 0) {
         free(array17->data);
         free(array17);
         DEBUG_FREE(array17->id);
      }
      array16->refC--;
      if(array16->refC == 0) {
         free(array16->data);
         free(array16);
         DEBUG_FREE(array16->id);
      }
   }
   else {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array24 = func34(&params0, loopsFactor);
      DEBUG_RETURN(array24->id);
      free(params0.data);
      array24->refC--;
      if(array24->refC == 0) {
         free(array24->data);
         free(array24);
         DEBUG_FREE(array24->id);
      }
   }
   array_t* array26;
   if (pCounter > 0) {
      array26 = vars->data[--pCounter];
      array26->refC++;
      DEBUG_COPY(array26->id);
   } else {
      array26 = (array_t*)malloc(sizeof(array_t));
      array26->size = 795;
      array26->refC = 1;
      array26->id = 26;
      array26->data = (unsigned int*)malloc(array26->size*sizeof(unsigned int));
      memset(array26->data, 0, array26->size*sizeof(unsigned int));
      DEBUG_NEW(array26->id);
   }
   return array26;
}

File: data/multilang_comparison/program_nim/src/func30.nim
proc func30*(vars: ptr ArrayTParam, PATH0: uint64, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   if (PATH0 and 1'u64) != 0:
      discard
      var array69: ptr ArrayT
      if pCounter > 0:
         dec pCounter
         array69 = vars.data[pCounter]
         inc array69.refC
         debugCopy(array69.id)
      else:
         array69 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
         array69.size = 237
         array69.refC = 1
         array69.id = 69
         array69.data = cast[ptr UncheckedArray[uint32]](alloc(array69.size * sizeof(uint32)))
         zeroMem(array69.data, array69.size * sizeof(uint32))
         debugNew(array69.id)
      let loopLimit23 = (50) div (4) + 1
      for loop23 in 0 ..< loopLimit23:
         discard
         for i in 0 ..< array69.size:
            dec array69.data[i]
         for i in 0 ..< array69.size:
            if array69.data[i] == 18:
               return array69
      var params0: ArrayTParam
      params0.size = 1
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      params0.data[0] = array69
      var array70: ptr ArrayT = func42(addr params0, getPath(), loopsFactor)
      debugReturn(array70.id)
      dealloc(params0.data)
      dec array70.refC
      if array70.refC == 0:
         dealloc(array70.data)
         dealloc(array70)
         debugFree(array70.id)
      dec array69.refC
      if array69.refC == 0:
         dealloc(array69.data)
         dealloc(array69)
         debugFree(array69.id)
   else:
      discard
      var params0: ArrayTParam
      params0.size = 0
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      var array93: ptr ArrayT = func34(addr params0, getPath(), loopsFactor)
      debugReturn(array93.id)
      dealloc(params0.data)
      dec array93.refC
      if array93.refC == 0:
         dealloc(array93.data)
         dealloc(array93)
         debugFree(array93.id)
   var array103: ptr ArrayT
   if pCounter > 0:
      dec pCounter
      array103 = vars.data[pCounter]
      inc array103.refC
      debugCopy(array103.id)
   else:
      array103 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
      array103.size = 933
      array103.refC = 1
      array103.id = 103
      array103.data = cast[ptr UncheckedArray[uint32]](alloc(array103.size * sizeof(uint32)))
      zeroMem(array103.data, array103.size * sizeof(uint32))
      debugNew(array103.id)
   return array103

